### PR TITLE
Fixes the login plugin event trigger and adds default login success event listener.

### PR DESCRIPTION
**Pull request by wshafer**
This fixes the login success process so login works out of the box.

This should be backwards compatible, but a test against your code base is recommended.

### CODE REVIEW

**Comment by wshafer** [comment]
Ping @jerv13  @rodmcnew

File: src/EventListener/Login.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        /** @var $request \Zend\Http\Request */
+        $request = $serviceManager->get('request');
+
+        $redirect = $request->getUri()->toString();
+
+        if (!empty($config['rcmPlugin']['RcmLogin']['defaultSuccessRedirect'])) {
+            $redirect = $config['rcmPlugin']['RcmLogin']['defaultSuccessRedirect'];
+        }
+
+        $redirect = $request->getQuery('redirect', $redirect);
```
Comment (rodmcnew) [inline src/EventListener/Login.php]: James and I looked over this. We can pull this PR if one thing is changed. Our current system doesn't allow ?redirect to take people off of the site. The code in this PR should be the same way. This prevents an attack where people put a fake version of your site in ?redirect so you think you are logging into one site then go to another.

If the query string redirect is modded to only allow relative URLs on the current site, we can pull this.

Comment (rodmcnew) [inline src/EventListener/Login.php]: Here is our current filtering on the redirect url 

```
public function parseRedirectUrl($url)
{
    return str_replace(['http://', 'https://', '//'], null, filter_var($url, FILTER_SANITIZE_URL));
}
```

**Comment by wshafer** [comment]
@rodmcnew - Good Catch!   I have added a filter to check for that and reworked the logic a bit.  See if that's more to your liking.

**Comment by rodmcnew** [comment]
This has been tagged in 0.5.0

**Pull request merged**